koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -881,65 +853,313 @@ def handler(self, src, target, opts=None):
+            yum_repourls=opts.get('yum_repourls', None),
+            branch=opts.get('git_branch', None),
+            push_url=opts.get('push_url', None),
+            arches=archlist,
+            koji_parent_build=opts.get('koji_parent_build'),
+            release=release_overwrite,
+            flatpak=flatpak,
+            signing_intent=opts.get('signing_intent', None),
+            compose_ids=opts.get('compose_ids', None),
+            skip_build=opts.get('skip_build', False),
+            triggered_after_koji_task=opts.get('triggered_after_koji_task', None)
+        )
+
+        results = []
+        semi_results = self.createContainer(**kwargs)
+        if semi_results is not None:
+            results = [semi_results]
+
+        self.logger.debug("Results: %r", results)
+
+        all_repositories = []
+        all_koji_builds = []
+
+        if not results:
+            return {
+                'repositories': all_repositories,
+                'koji_builds': all_koji_builds,
+                'build': 'skipped',
+            }
+
+        for result in results:
+            try:
+                repository = result.get('repositories')
+                all_repositories.extend(repository)
+            except Exception as error:
+                self.logger.error("Failed to merge list of repositories "
+                                  "%r. Reason (%s): %s", repository,
+                                  type(error), error)
+            koji_build_id = result.get('koji_build_id')
+            if koji_build_id:
+                all_koji_builds.append(koji_build_id)
+
+        return {
+            'repositories': all_repositories,
+            'koji_builds': all_koji_builds,
+        }
+
+
+class BuildSourceContainerTask(BaseContainerTask):
+    # Start builds via osbs
+    Methods = ['buildSourceContainer']
+    # Same value as for regular 'build' method.
+    _taskWeight = 2.0
+
+    # JSON Schema definition for koji buildSourceContainer task parameters
+    # Used to validate arguments passed to the handler() method of this class
+    PARAMS_SCHEMA = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "description": "Parameters for a koji buildSourceContainer task.",
+
+        "type": "array",
+        "items": [
+            {
+                "type": "string",
+                "description": "Build target."
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources"
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources"
```

Comment:
Could we add an "examples" field here with some samples of possible NVR values? we could refer to this schema later in the docs

Reply:
sure